src/lib/CoreInstance.cc

```diff
@@ -0,0 +1,240 @@
+
+  return;
+}
+
+void CoreInstance::createProcess(span<char> instructions) {
+  process_ =
+      std::make_unique<simeng::kernel::LinuxProcess>(instructions, config_);
+
+  // Raise error if created process is not valid
+  if (!process_->isValid()) {
+    std::cerr << "Could not create process based on supplied instruction span"
+              << std::endl;
+    exit(1);
+  }
+
+  // Create the process memory space from the generated process image
+  createProcessMemory();
+
+  // Create the OS kernel with the process
+  kernel_.createProcess(*process_.get());
+
+  return;
+}
+
+std::shared_ptr<simeng::MemoryInterface>
+CoreInstance::createL1InstructionMemory(const simeng::MemInterfaceType type) {
+  // Currently, only a flat memory interface can be used for the instruction
+  // memory
+  if (type != simeng::MemInterfaceType::Flat) {
+    std::cerr << "Incompatible instruction memory interface type requested"
+              << std::endl;
+    exit(1);
+  }
+
+  // Create a L1I cache instance based on type supplied
+  if (type == simeng::MemInterfaceType::Flat) {
+    instructionMemory_ = std::make_shared<simeng::FlatMemoryInterface>(
+        processMemory_, processMemorySize_);
+  } else if (type == simeng::MemInterfaceType::Fixed) {
+    instructionMemory_ = std::make_shared<simeng::FixedLatencyMemoryInterface>(
+        processMemory_, processMemorySize_,
+        config_["L1-Cache"]["Access-Latency"].as<uint16_t>());
+  }
+
+  return instructionMemory_;
+}
+
+void CoreInstance::setL1InstructionMemory(
+    std::shared_ptr<simeng::MemoryInterface> memRef) {
+  // Set the L1I cache instance to use
+  instructionMemory_ = memRef;
+  return;
+}
+
+std::shared_ptr<simeng::MemoryInterface> CoreInstance::createL1DataMemory(
+    const simeng::MemInterfaceType type) {
+  // Currently, if the core in use is emulation or in-order, only a flat data
+  // memory interface can be used
+  if (mode_ == SimulationMode::Emulation ||
+      mode_ == SimulationMode::InOrderPipelined) {
+    if (type != simeng::MemInterfaceType::Flat) {
+      std::cerr << "Incompatible instruction memory interface type requested"
+                << std::endl;
+      exit(1);
+    }
+  }
+
+  // Create a L1D cache instance based on type supplied
+  if (type == simeng::MemInterfaceType::Flat) {
+    dataMemory_ = std::make_shared<simeng::FlatMemoryInterface>(
+        processMemory_, processMemorySize_);
+  } else if (type == simeng::MemInterfaceType::Fixed) {
+    dataMemory_ = std::make_shared<simeng::FixedLatencyMemoryInterface>(
+        processMemory_, processMemorySize_,
+        config_["L1-Cache"]["Access-Latency"].as<uint16_t>());
+  }
+
+  return dataMemory_;
+}
+
+void CoreInstance::setL1DataMemory(
+    std::shared_ptr<simeng::MemoryInterface> memRef) {
+  // Set the L1D cache instance to use
+  dataMemory_ = memRef;
+  return;
+}
+
+std::shared_ptr<simeng::Core> CoreInstance::createCore() {
+  // Ensure all appropriate creation functions have been called
```

Comment:
Are the prerequisites for `createCore` documented anywhere, so that the developer knows what order to call these functions in?

Reply:
No as of yet. It's too early in the development pipeline of this new feature to consider writing documentation.